### PR TITLE
NN-3069 return extra transaction info

### PR DESCRIPTION
**Pull request by JamieDraperUK**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/service/OffenderTransactionHistoryService.java [hunk 1]
```diff
@@ -2,73 +2,69 @@
...
             boolean isAccountCodeExists = AccountCode.exists(accountCodeOpl.get());
             checkState(isAccountCodeExists, "Unknown account-code " + accountCodeOpl.get());
         }
 
-        return getSortedHistories(offenderNo, accountCodeOpl, fromDate, toDate).stream()
-                .map(h -> Pair.of(h, apiCurrency))
-                .map(OffenderTransactionHistoryTransformer::transform)
-                .collect(Collectors.toList());
+        return accountCodeOpl.map(AccountCode::byCodeName)
+            .filter(Optional::isPresent)
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/hmpps/prison/service/OffenderTransactionHistoryService.java:60]: Do you need a test for this filtering?

File: src/main/java/uk/gov/justice/hmpps/prison/service/OffenderTransactionHistoryService.java [hunk 2]
```diff
@@ -2,73 +2,69 @@
...
 
-    private String apiCurrency;
-    private OffenderTransactionHistoryRepository historyRepository;
-    private OffenderRepository offenderRepository;
+    private final String apiCurrency;
+    private final OffenderTransactionHistoryRepository historyRepository;
 
     public OffenderTransactionHistoryService(@Value("${api.currency:GBP}") final String currency,
-                                             final OffenderTransactionHistoryRepository historyRepository,
-                                             final OffenderRepository offenderRepository) {
```
Comment (JamieDraperUK) [inline src/main/java/uk/gov/justice/hmpps/prison/service/OffenderTransactionHistoryService.java:41]: Not used.

File: src/main/java/uk/gov/justice/hmpps/prison/service/OffenderTransactionHistoryService.java [hunk 3]
```diff
@@ -89,19 +85,21 @@ private void validate(final String offenderNo,
...
-                .map(AccountCode::byCodeName)
-                .filter(Optional::isPresent)
-                .map(optionalCode -> optionalCode.get().code)
-                .map(code -> historyRepository.findForGivenAccountType(offenderNo, code, fromDate, toDate))
-                .orElse(historyRepository.findForAllAccountTypes(offenderNo, fromDate, toDate));
-
-        Collections.sort(histories, TRANSACTION_HISTORY_SORTING_POLICY);
-
-        return histories;
+    public OffenderTransactionHistoryDto transform(final OffenderTransactionHistory offenderTransactionHistory) {
```
Comment (JamieDraperUK) [inline src/main/java/uk/gov/justice/hmpps/prison/service/OffenderTransactionHistoryService.java:88]: Moved closer to usage and resolved the hard coded currency value.

File: src/main/java/uk/gov/justice/hmpps/prison/service/OffenderTransactionHistoryService.java (hunk 1 above)
Comment (JamieDraperUK) [reply src/main/java/uk/gov/justice/hmpps/prison/service/OffenderTransactionHistoryService.java:60]: These is an existing test that covers both routes. One being when the account code is available and the other when not. I've added a couple of tests for mapping and these also make sure that both paths are tested.

**Review by jamesak-oncloud9 — APPROVED** [review]

**Pull request merged**